### DISCUSSION TITLE
1.4.1-beta2

### DISCUSSION THREAD

**Discussion by ilyhalight** [ilyhalight/voice-over-translation#286] [Обновления]
⚠ Это бета версия и она может работать нестабильно. Обо всех проблемах рекомендуется сразу же сообщать в [Issues](https://github.com/ilyhalight/voice-over-translation/issues)

## Список изменений
- Исправлено изменение процента громкости видео при изменение громкости через слайдер ютуба
- Переработано получение и установка громкости видео
- Исправлен сброс установленной громкости звука при открытие нового видео

**Full Changelog**: https://github.com/ilyhalight/voice-over-translation/compare/1.4.1-beta...1.4.1-beta2

<hr /><em>This discussion was created from the release <a href='https://github.com/ilyhalight/voice-over-translation/releases/tag/1.4.1-beta2'>1.4.1-beta2</a>.</em>

**Comment by NullVerdict** [comment]
Есть небольшой баг, связанный с субтитрами. Если включить субтитры на одном видео, а потом их выключить и перейти на другое видео, то функция “Переводить при открытии” не будет работать.

**Reply by NullVerdict** [reply]
Также после внедрения субтитров, они вызваются слишком часто. Разве нельзя просто сделать переменную и сохранить их там, а не вызывать по 1000 раз? Тоже самое с определением языка видео. Он может отправить запрос на получение языка, во время навигации на ютубе. 
![изображение](https://github.com/ilyhalight/voice-over-translation/assets/24498484/2e6e88ba-9afd-4930-b42c-179cec8b3640)
![изображение](https://github.com/ilyhalight/voice-over-translation/assets/24498484/b585d2c2-21f0-4c49-831a-dedcf3b88a71)

**Reply by xirthas** [reply]
>Также после внедрения субтитров, они вызваются слишком часто. Разве нельзя просто сделать переменную и сохранить их там, а не вызывать по 1000 раз?

Список субтитров должен запрашиваться один раз для видео при открытии меню(яБраузер запрашивает каждый раз при открытии меню), собственно, так у меня и происходит. 1 запрос это 1 OPTIONS и 1 POST метод. Но если открыть меню, закрыть и снова открыть до того, как завершится первый запрос, то отправится еще один, что, вероятно, у тебя и произошло.

**Reply by NullVerdict** [reply]
> > Также после внедрения субтитров, они вызваются слишком часто. Разве нельзя просто сделать переменную и сохранить их там, а не вызывать по 1000 раз?
> 
> Список субтитров должен запрашиваться один раз для видео при открытии меню(яБраузер запрашивает каждый раз при открытии меню), собственно, так у меня и происходит. 1 запрос это 1 OPTIONS и 1 POST метод. Но если открыть меню, закрыть и снова открыть до того, как завершится первый запрос, то отправится еще один, что, вероятно, у тебя и произошло.

Проблема в том, что каждый раз, когда ты открываешь новое видео, отправляется больше новых запросов на субтитры. Например, если я включил субтитры на одном видео, а потом перешел на другое, то функция “Переводить при открытии” перестает работать. И чем больше я пытаюсь активировать субтитры, тем больше запросов я посылаю. Получается, что я нечаянно организовал спам-атаку на сайт из-за этого бага.

**Comment by NullVerdict** [comment]
Также на [этом](https://youtu.be/UJeSWbR6W04) видео есть баг. Язык определяется как en-US
